spec/tapioca/gem/pipeline_spec.rb

```diff
@@ -3271,6 +3271,51 @@ def non_abstract_children; end
       assert_equal(output, compile)
     end
 
+    it 'compiles constants of generic types' do
+      add_ruby_file("generic.rb", <<~RUBY)
+        module GenericInterface
+          extend T::Generic
+
+          interface!
+
+          Parameter = type_member
+        end
+
+        class Concrete
+          extend T::Generic
+
+          include GenericInterface
+          Parameter = type_member
+        end
+
+        GENERIC_CONSTANT = T.let(
+          Concrete[Numeric].new,
```

Comment:
```suggestion
          Concrete.new,
```

The `#[]` should not be used at runtime. What is funny is that sorbet even show an error when you try to do that in some kind of expressions like `if`.

Reply:
But given that, is this what causes the error you are trying to fix?